test/engine/TextIndexScanForWordTest.cpp

```diff
@@ -63,27 +39,73 @@ TEST(TextIndexScanForWord, WordScanBasic) {
 
-  ASSERT_EQ(combineToString("\"he failed the test\"", "test"),
-            combineToString(getTextRecordFromResultTable(qec, result, 0),
-                            getWordFromResultTable(qec, result, 0)));
-  ASSERT_EQ(combineToString("\"testing can help\"", "testing"),
-            combineToString(getTextRecordFromResultTable(qec, result, 1),
-                            getWordFromResultTable(qec, result, 1)));
-  ASSERT_EQ(combineToString("\"the test on friday was really hard\"", "test"),
-            combineToString(getTextRecordFromResultTable(qec, result, 2),
-                            getWordFromResultTable(qec, result, 2)));
+  ASSERT_EQ(h::combineToString("\"he failed the test\"", "test"),
+            h::combineToString(h::getTextRecordFromResultTable(qec, result, 0),
+                               h::getWordFromResultTable(qec, result, 0)));
+  ASSERT_EQ(h::combineToString("\"testing can help\"", "testing"),
+            h::combineToString(h::getTextRecordFromResultTable(qec, result, 1),
+                               h::getWordFromResultTable(qec, result, 1)));
+  ASSERT_EQ(
+      h::combineToString("\"the test on friday was really hard\"", "test"),
+      h::combineToString(h::getTextRecordFromResultTable(qec, result, 2),
+                         h::getWordFromResultTable(qec, result, 2)));
+}
 
-  TextIndexScanForWord s3{qec, Variable{"?text1"}, "test"};
+TEST(TextIndexScanForWord, WordScanBasic) {
+  auto qec = getQec(kg, true, true, true, 16_B, true);
+
+  TextIndexScanForWord s1{qec, Variable{"?text1"}, "test"};
 
-  ASSERT_TRUE(s1.asStringImpl(0) != s3.asStringImpl(0));
-  ASSERT_EQ(s3.getResultWidth(), 1);
+  ASSERT_EQ(s1.getResultWidth(), 1);
 
-  result = s3.computeResultOnlyForTesting();
+  auto result = s1.computeResultOnlyForTesting();
   ASSERT_EQ(result.width(), 1);
   ASSERT_EQ(result.size(), 2);
 
   ASSERT_EQ("\"he failed the test\"",
-            getTextRecordFromResultTable(qec, result, 0));
+            h::getTextRecordFromResultTable(qec, result, 0));
   ASSERT_EQ("\"the test on friday was really hard\"",
-            getTextRecordFromResultTable(qec, result, 1));
+            h::getTextRecordFromResultTable(qec, result, 1));
+
+  TextIndexScanForWord s2{qec, Variable{"?text1"}, "testing"};
+
+  ASSERT_EQ(s2.getResultWidth(), 1);
+
+  result = s2.computeResultOnlyForTesting();
+  ASSERT_EQ(result.width(), 1);
+  ASSERT_EQ(result.size(), 1);
+
+  ASSERT_EQ("\"testing can help\"",
+            h::getTextRecordFromResultTable(qec, result, 0));
+}
+
+TEST(TextIndexScanForWord, CacheKey) {
+  auto qec = getQec(kg, true, true, true, 16_B, true);
+
+  TextIndexScanForWord s1{qec, Variable{"?text1"}, "test*"};
+  TextIndexScanForWord s2{qec, Variable{"?text2"}, "test*"};
+  ASSERT_EQ(s1.getCacheKeyImpl(), s2.getCacheKeyImpl());
+
+  TextIndexScanForWord s3{qec, Variable{"?text1"}, "test"};
+  ASSERT_TRUE(s1.getCacheKeyImpl() != s3.getCacheKeyImpl());
```

Comment:
Use ASSERT_NE here.
And maybe for completeness the case with var and word differently.